shared/src/TwinCollection.cs

```diff
@@ -285,14 +296,33 @@ private void TryClearMetadata(string propertyName)
         }
 
         /// <summary>
-        /// Clear metadata out of the collection
+        /// Clears metadata out of the properties collection.
         /// </summary>
+        /// <remarks>
+        /// This will only clear the metadata from the properties collection but will not change the <see cref="TwinCollection"/> metadata collection object. This allows you to still use methods such as <see cref="GetMetadata"/>. If you need to remove all metadata, please use <see cref="ClearAllMetadata"/>.
+        /// </remarks>
         public void ClearMetadata()
         {
             TryClearMetadata(MetadataName);
             TryClearMetadata(LastUpdatedName);
             TryClearMetadata(LastUpdatedVersionName);
             TryClearMetadata(VersionName);
         }
+
+        /// <summary>
+        /// Clears all metadata out of the twin collection as well as all of the metadata from the properties collection.
+        /// </summary>
+        /// <remarks>
+        /// This will remove all metadata from the <see cref="TwinCollection"/> metadata object as well as the metadata for all of the properties collection. The difference from the <see cref="ClearMetadata"/> method is it will also clear the underlying metadata container which will affect methods such as <see cref="GetMetadata"/> and <see cref="GetLastUpdatedVersion"/>.
+        /// This method would be useful if you are performing any operations that require <see cref="TryGetMemberInternal(string, out object)"/> to return a <see cref="JToken"/> regardless of the client you are using.
+        /// </remarks>
+        public void ClearAllMetadata()
+        {
+            ClearMetadata();
+            // GitHub Issue: https://github.com/Azure/azure-iot-sdk-csharp/issues/1971
```

Comment:
I do like the comments on line 323 and 324 but does adding a link to github issue here help?

Reply:
I do think it helps since we might need to go back to it at some point in time "Why does this clear both" type of question.